### PR TITLE
Gt/clean celllab code

### PR DESCRIPTION
**Pull request by gregtucker**
Cleans up celllab_cts.py and related, removing unused Python functions that were translated to Cython, and some older functions that were superseded.

### CODE REVIEW

**Comment by gregtucker** [comment]
Improved the coverage of the core CA files, so hopefully the coverage test will now pass too.

File: landlab/components/flow_accum/flow_accumulator.py [hunk 1]
```diff
@@ -1136,7 +1136,7 @@ def accumulate_flow(self, update_flow_director=True):
                 self.depression_finder.map_depressions()
 
                 # if FlowDirectorSteepest is used, update the link directions
-                if self.flow_director._name is "FlowDirectorSteepest":
+                if self.flow_director._name == "FlowDirectorSteepest":
                     self.flow_director._determine_link_directions()
```
Comment (kbarnhart) [inline landlab/components/flow_accum/flow_accumulator.py]: @gregtucker I think all these comparison changes improve the code, but perhaps they belong in another PR. @mcflugen what do you think?

Comment (gregtucker) [reply landlab/components/flow_accum/flow_accumulator.py]: Travis tests were failing because of flake8 linter issues, including the above, so in the interest of expediency I went ahead and fixed them.
